apps/client/src/context/authContext.tsx

```diff
@@ -11,32 +12,70 @@ import {
 type AuthContextType = {
   accessToken: string | null;
   setAccessToken: (token: string | null) => void;
+  refreshToken: string | null;
+  setRefreshToken: (token: string | null) => void;
+  refreshTokenPair: () => Promise<void>;
 };
 
 const AuthContext = createContext<AuthContextType | null>(null);
 
 export const AuthContextProvider = ({
   children,
 }: { children: React.ReactNode }) => {
-  const [accessToken, _setAccessToken] = useState<string | null>(null);
+  const [accessToken, setAccessToken] = useState<string | null>(null);
+  const [refreshToken, setRefreshToken] = useState<string | null>(null);
 
   useEffect(() => {
     const token = localStorage.getItem("accessToken");
-    _setAccessToken(token);
-  }, [_setAccessToken]);
+    if (token) setAccessToken(token);
+  }, [setAccessToken]);
 
-  const setAccessToken = useCallback((token: string | null) => {
-    if (token === null) {
+  useEffect(() => {
+    const token = localStorage.getItem("refreshToken");
+    if (token) setRefreshToken(token);
+  }, [setRefreshToken]);
+
+  useEffect(() => {
+    if (accessToken === null) {
       localStorage.removeItem("accessToken");
       return;
     }
+    localStorage.setItem("accessToken", accessToken);
+  }, [accessToken]);
+
+  useEffect(() => {
+    if (refreshToken === null) {
+      localStorage.removeItem("refreshToken");
+      return;
+    }
+    localStorage.setItem("refreshToken", refreshToken);
+  }, [refreshToken]);
```

Comment:
We probably should opt to use some external hooks library because this is unmaintainable and this code has many culprits. 

Check how `usehooks` library handles their store in sync:
https://github.com/uidotdev/usehooks/blob/dfa6623fcc2dcad3b466def4e0495b3f38af962b/index.js#L616-L652


This can be a TODO

Reply:
I'm not sure if the `useLocalStorage` hook will play very nicely with the current structure of AuthContextProvider, but I think we can make it work. The API looks way nicer than calling the raw `localStorage` functions :^)